integration/spark/shared/src/main/java/io/openlineage/spark/agent/hooks/DatabricksJobEventBuilderHook.java

```diff
@@ -0,0 +1,76 @@
+/*
+/* Copyright 2018-2022 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+package io.openlineage.spark.agent.hooks;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.agent.util.DatabricksUtils;
+import io.openlineage.spark.api.OpenLineageContext;
+
+public class DatabricksJobEventBuilderHook implements RunEventBuilderHook {
+
+  private final OpenLineageContext openLineageContext;
+
+  public DatabricksJobEventBuilderHook(OpenLineageContext openLineageContext) {
+    this.openLineageContext = openLineageContext;
+  }
+
+  @Override
+  public void preBuild(OpenLineage.RunEventBuilder runEventBuilder) {
+    if (!DatabricksUtils.isRunOnDatabricksPlatform(openLineageContext)) {
+      return;
+    }
+
+    OpenLineage.RunEvent runEvent = runEventBuilder.build();
+    OpenLineage.Job job = runEvent.getJob();
+
+    StringBuilder jobNameBuilder = new StringBuilder();
+    jobNameBuilder
+        .append(extractClusterId(DatabricksUtils.getWorkspaceUrl(openLineageContext).get()))
+        .append(":")
+        .append(job.getName().replace("databricks_shell.", "")); // remove default job_name
```

Comment:
More of a question really but how are others using the job name? Is it mainly used for display purposes inside of Marquez or other tools?

We use the notebook name (spark.databricks.notebook.path as part of environment properties) or databricks job/task name but the job / task name requires additional API calls.

Reply:
There are two reasons behind modifying job name. For example, for a job name `databricks_shell.append_exec_v1`
1. It is user unfriendly identifier
2. It gets repeated whenever data is added to any table. 
The reason is mainly meaningful display purpose.

Were you able to extract notebook path and name from Spark properties? I was not able to find that.

Reply:
We're doing it in the DatabricksEnvironmentFacetBuilder!
https://github.com/OpenLineage/OpenLineage/blob/359e338699bc0c2e57f727c85a7193a74bdf2dc5/integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/DatabricksEnvironmentFacetBuilder.java#L58